components/datadog/agentparams/filepermissions/permissions_nix.go

```diff
@@ -0,0 +1,82 @@
+package filepermissions
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+	"github.com/DataDog/test-infra-definitions/common"
+)
+
+type UnixPermissionsOption = func(*UnixPermissions) error
+
+// UnixPermissions represents the owner, group, and permissions of a file.
+// Permissions are represented by a string (should be an octal). There is no check on the octal format.
+type UnixPermissions struct {
+	Owner       string
+	Group       string
+	Permissions string
+}
+
+var _ FilePermissions = (*UnixPermissions)(nil)
+
+// NewUnixPermissions creates a new UnixPermissions object and applies the given options.
+func NewUnixPermissions(options ...UnixPermissionsOption) optional.Option[FilePermissions] {
+	p, err := common.ApplyOption(&UnixPermissions{}, options)
+
+	if err != nil {
+		return optional.NewNoneOption[FilePermissions]()
+	}
+	return optional.NewOption[FilePermissions](p)
+}
+
+// SetupPermissionsCommand returns a command that sets the owner, group, and permissions of a file.
+func (p *UnixPermissions) SetupPermissionsCommand(path string) string {
+	var commands []string
+
+	if p.Owner != "" {
+		commands = append(commands, fmt.Sprintf("chown %s %s", p.Owner, path))
+	}
+
+	if p.Group != "" {
+		commands = append(commands, fmt.Sprintf("chgrp %s %s", p.Group, path))
+	}
+
+	if p.Permissions != "" {
+		commands = append(commands, fmt.Sprintf("chmod %s %s", p.Permissions, path))
+	}
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏You might need to prefix every command with `sudo` ?

Otherwise I guess with some combination it would fail, eg. first change the user and group to `root`, then try to change the permissions

Reply:
You're right, I've just tested this scenario and it failed without `sudo` 👍

Reply:
Actually my test was failing for another reason 😓  so i think the first sudo does the job. I'll keep all `sudo` anyway as it does not hurt I believe

Reply:
I actually didn't realize you were running all the commands in a subshell, you really didn't need it 🤦